### PR TITLE
feat: options taxation per DGT V0137-23 (Art. 37.1.m LIRPF)

### PR DESCRIPTION
**Pull request by GeiserX**
## Summary
- Implements complete options lifecycle handling with 3 scenarios per V0137-23:
  - **Expiration**: option expires worthless (full premium loss for buyer, gain for writer)
  - **Early close**: opposite trade closes position (premium difference as G/P)
  - **Exercise/Assignment**: option P&L recorded separately + underlying acquired/delivered at strike
- Parses `OptionEAE` XML section from IBKR Flex Query (exercise/assignment/expiration events)
- Propagates option fields (`putCall`, `strike`, `expiry`, `underlyingSymbol`) through `Lot` → `FifoDisposal`
- Shows option scenario info in CSV, PDF, and web operations annex
- Fixes pre-existing lint errors in `casilla-detail.ts` and `profile.test.ts`

## Legal basis
- Art. 37.1.m) LIRPF: options/futures as ganancias patrimoniales (base del ahorro)
- DGT V0137-23: consulta vinculante on option exercise with physical delivery
- Art. 33.5.f: anti-churning correctly exempts OPT (already implemented, now tested)

## Test plan
- [x] 12 new tests covering all 3 scenarios (buyer/writer × expiration/close/exercise)
- [x] Edge cases: no prior lots, partial expiration, field propagation
- [x] All 866 tests passing locally
- [x] Typecheck clean
- [x] Lint clean

## Summary by CodeRabbit

* **New Features**
  * Option exercises, assignments, and expirations are now properly processed and reported instead of generating warnings.
  * Option metadata (put/call, strike, expiry, underlying symbols) is tracked and displayed throughout reports.

* **Improvements**
  * CSV exports now include option categorization and scenario columns.
  * PDF reports display option details with shorthand notation.
  * Web interface labels options and displays scenario badges with relevant details.

* **Tests**
  * Added comprehensive test coverage for option taxation scenarios.

### CODE REVIEW

File: src/engine/fifo.ts [hunk 1]
```diff
@@ -590,6 +601,365 @@ export class FifoEngine {
...
+          acquireEcbRate: lot.ecbRate,
+          assetCategory: "OPT",
+          washSaleBlocked: false,
+          optionScenario: "exercise",
+          putCall: ex.putCall,
+          strike: ex.strike,
+          expiry: ex.expiry,
+          underlyingSymbol: ex.underlyingSymbol,
+          underlyingIsin: ex.underlyingIsin,
+        });
```
[the remark this answers is not in the document]
Comment (mariiio81) [reply src/engine/fifo.ts:814]: This is an incomplete revision. The V0137-23 also states that the underlying is adquired at market price, not at strike price.

**Comment by mariiio81** [comment]
This is an incomplete revision. The V0137-23 also states that the underlying is adquired at market price, not at strike price.

**Comment by GeiserX** [comment]
@mariiio81 muchas gracias por decirme, lo voy a revisar
